synfig-studio/src/gui/mainwindow.cpp

```diff
@@ -262,6 +274,18 @@ MainWindow::toggle_show_toolbar()
 	}
 }
 
+void
+MainWindow::appmenu_toggle_show_toolbar()
```

Comment:
You can simplify this by creating helper method:
```
void change_action_state(const std::string& name, bool active) {
	if (auto action = App::instance()->lookup_action(name)) {
		action->change_state(active);
	}
}
```

After that you need only one line to switch it to correct state:
```
void
MainWindow::toggle_show_toolbar()
{
	App::enable_mainwin_toolbar = !App::enable_mainwin_toolbar;
	change_action_state("show-toolbar", App::enable_mainwin_toolbar);
	...
}
```

`App::enable_mainwin_toolbar` already contains the correct state at this point, so you only need to pass it to the action.

Same with MainWindow::toggle_show_menubar:
```
void
MainWindow::toggle_show_menubar()
{
	Gtk::Widget* menubar = App::ui_manager()->get_widget("/menubar-main");
	App::enable_mainwin_menubar = !App::enable_mainwin_menubar;
	change_action_state("show-menubar", App::enable_mainwin_menubar);
	...
}
```

Reply:
@ice0 thank you good idea. What do you think about `toggle_show_menu_bar`?  There is nothing to toggle anymore on Macos since the menubar is now not a widget and is in the native `macos` menubar.

Reply:
@nickleus27 
I need to check Linux/Windows behavior. I will answer later.